File: kattis/jumbled_communication.pl
% https://open.kattis.com/problems/communication
% 2s
% 2.1 Easy


main :-
    readln([_|T],end_of_file),
    s(T),
    nl.
s([]).
s([N|T]) :-
    between(0,255,I),
    N =:= (I xor(I<<1)) mod 256,
    format("~d ",[I]),
    s(T).

/*
% Compressed: 123 chars. Top 10 place 5!
main:-readln([_|T],end_of_file),s(T),nl.
s([]). s([N|T]):-between(0,255,I),N=:=(I xor(I<<1))mod 256,format("~d ",[I]),s(T).
*/

File: kattis/jumbled_communication.py
# https://open.kattis.com/problems/communication
# 2s
# 2.1 Easy

# 134 chars: Should be in place 10 in Top 10, but perhaps they
# don't place the same person twice in the top list
# (my Prolog program is in place 5 with 123 chars).
# And it's not in Top 10 for Python3 either, should be
# in place 8.
# 
import sys
for x in[int(i)for i in sys.stdin.readlines()[1::][0].split()]:
 for i in range(256):
  if x==i^(i<<1)%256:print(i,end=" ")
